File: Top 50 Coding Interview Questions/20-Write the code to for Armstrong number/Armstrongnumber.c
#include<stdio.h>
#include<math.h>

// Armstrong number is any number following the given rule
// abcd... = a^n + b^n + c^n + d^n + ...
// Where n is the order(length/digits in number)

// Example = 153 (order/length = 3)
// 153 = 1^3 + 5^3 + 3^3 = 1 + 125 + 27 = 153

// Example = 1634 (order/length = 4)
// 1634 = 1^4 + 6^4 + 3^4 + 4^4 = 1 + 1296 + 81 + 256 = 1634

// number of digits in a number is order

int order(int x)
{
    int len = 0;
    while(x)
    {
        len++;
        x = x/10;
    }
    return len;
}

int armstrong(int num, int len){
    int sum = 0, temp, digit;
    temp = num;

    //loop to extarct digit, find power & add to sum
    while(temp != 0)
    {
        //extract digit
        digit = temp % 10;

        //add poer to sum
        sum = sum + pow(digit, len);
        temp /= 10;
    };
     
     return num == sum;
}

//Driver code
int main(){
    int num, len;

    printf("Enter a Number: ");
    scanf("%d", &num);

    //function to grt order(length)
    len = order(num);

    //check if Armstrong
    if(armstrong(num, len))
       printf("%d is Armstrong", num);
    else
       printf("%d is not Aramstrong", num);

}

File: Top 50 Coding Interview Questions/20-Write the code to for Armstrong number/Armstrongnumber.py
# Function to check if a number is an Armstrong number
def is_armstrong(num):
    # Convert the number to a string to find the number of digits
    num_str = str(num)
    digits = len(num_str)

    # Calculate the sum of each digit raised to the power of the number of digits
    sum_of_powers = sum(int(digit) ** digits for digit in num_str)

    # Check if the sum is equal to the original number
    return sum_of_powers == num

# Main function to execute the program
def main():
    # Prompt user for input
    num = int(input("Enter a number: "))

    # Check if the number is an Armstrong number and display the result
    if is_armstrong(num):
        print(f"{num} is an Armstrong number.")
    else:
        print(f"{num} is not an Armstrong number.")

# Execute the main function
if __name__ == "__main__":
    main()
